zluda/src/impl/dark_api.rs

```diff
@@ -62,6 +62,31 @@ impl CudaDarkApi for CudaDarkApiZluda {
         device::primary_ctx_get(pctx, hip_dev).into_cuda()
     }
 
+    unsafe extern "system" fn set_device_flags(
+        dev: cuda_types::CUdevice,
+        flags: c_uint,
+    ) -> CUresult {
+        use hip_runtime_sys::*;
+        let hip_dev = FromCuda::from_cuda(dev);
+        let mut active_dev = 0i32;
+        hipGetDevice(&mut active_dev as _);
```

Comment:
The return value `hipError_t` of `hipGetDevice` must be used.

Reply:
Let's assume hipGetDevice failed: what behavior do you expect? Abort the whole function? Resume without changing device?

Reply:
CUDA Runtime API returns `cudaErrorInvalidValue` in such situation.